tasks/java.yml

```diff
@@ -2,6 +2,8 @@
 - name: "Check if Java is installed"
   shell: which java
   register: check_java
+  failed_when: check_java.rc not in  (0, 1)
+  changed_when: False
```

Comment:
I wonder why that could be happening. I've not been able to get a return code of 0 for something unless it is found in $PATH.  It seems pretty straightforward:

    jay@jays:~/$ command -v java; echo $?
/usr/bin/java
    0
    jay@jays:~/$ command -v javaddeeee; echo $?
    1

I'd prefer we get to the bottom of this so we can ensure the behavior of our playbook is consistent and reliable.  Can you check again and/or show me more debug output with `command -v` via something like:

    name: "Debug the thing"
    debug: var=check_java